pkg/util/rolloutblock/blocks.go

```diff
@@ -63,18 +61,34 @@ func BlocksRollout(rolloutBlockLister shipperlisters.RolloutBlockLister, obj met
 	}
 }
 
-func ValidateOverrides(overrides ObjectNameList) error {
-	if len(overrides) == 0 {
+func ProcessBlocks(existingBlocks ObjectNameList, overrides ObjectNameList) error {
+	effectiveBlocks := existingBlocks.Diff(overrides)
+	if len(effectiveBlocks) == 0 {
 		return nil
+	} else {
```

Comment:
Suggested edit: drop the else statement as the nil-case return returns instantly.